neuralmonkey/decoders/sequence_labeler.py

```diff
@@ -52,64 +62,36 @@ def target_tokens(self) -> tf.Tensor:
 
     @tensor
     def train_targets(self) -> tf.Tensor:
-        return self.vocabulary.strings_to_indices(self.target_tokens)
+        return self.vocabulary.strings_to_indices(
+            self.dataset[self.data_id])
 
     @tensor
     def train_mask(self) -> tf.Tensor:
         return sentence_mask(self.train_targets)
 
-    @property
-    def rnn_size(self) -> int:
-        return int(self.encoder.temporal_states.get_shape()[-1])
-
-    @tensor
-    def decoding_w(self) -> tf.Variable:
-        return get_variable(
-            name="state_to_word_W",
-            shape=[self.rnn_size, len(self.vocabulary)])
-
     @tensor
-    def decoding_b(self) -> tf.Variable:
-        return get_variable(
-            name="state_to_word_b",
-            shape=[len(self.vocabulary)],
-            initializer=tf.zeros_initializer())
+    def concatenated_inputs(self) -> tf.Tensor:
```

Comment:
Kdyz budu mit vice enkoderu, co pracuji nad ruzne dlouhymi sekvencemi, tak to na tom concatu spadne, ne?
Nemela by se takova situace resit spise pres FactoredSequence/FactoredEncoder?

Reply:
to je pravda